### PR TITLE
Optimise gas usage during the upgrade process 

### PR DESCRIPTION
**Pull request by ctindogaru**
This PR is trying to address https://github.com/near-daos/sputnik-dao-contract/issues/135.

### CODE REVIEW

File: sputnikdao-factory2/src/factory_manager.rs [hunk 1]
```diff
@@ -7,13 +7,13 @@ use near_sdk::serde_json;
 use near_sdk::{env, sys, AccountId, Balance, CryptoHash, Gas};
 
 /// Gas spent on the call & account creation.
-const CREATE_CALL_GAS: Gas = Gas(75_000_000_000_000);
```
Comment (ctindogaru) [inline sputnikdao-factory2/src/factory_manager.rs:10]: This was absurd. A simple `new` call to create a DAO does not require more than `5_000_000_000_000` gas. However, I've doubled it to `10_000_000_000_000` just to be safe.

File: sputnikdao-factory2/src/factory_manager.rs [hunk 2]
```diff
@@ -7,13 +7,13 @@ use near_sdk::serde_json;
...
+const CREATE_CALL_GAS: Gas = Gas(50_000_000_000_000);
```
[the remark this answers is not in the document]
Comment (ctindogaru) [reply sputnikdao-factory2/src/factory_manager.rs]: Are you not happy with 50 or with a value lower than 50?

I'm not sure if a larger policy would consume more gas? It makes sense to consume more storage, but is the gas affected as well? Shouldn't be and if it is, I think the impact is pretty low...

[the remark this answers is not in the document]
Comment (ctindogaru) [reply sputnikdao-factory2/src/factory_manager.rs]: I'm not 100% sure here either, you made a valid point

[the remark this answers is not in the document]
Comment (TrevorJTClarke) [reply sputnikdao-factory2/src/factory_manager.rs]: Been having a solid think on this:
As I have deployed DAOs from other DAOs and within other batch calls, even the previous 75Tgas was plenty to get through. Even at 75Tgas, i have seen an avg of 5-25Tgas based on the policy tests i've done. This is not the maximum size a policy can be however - so maybe a little R&D is needed to find the absolute policy size at creation. 
Here's what I'd like to see:
Can you create a test case with a large amount of groups AND/OR a large amount of council accounts?
That should give us a clear indication of how big the gas needs to be.

[the remark this answers is not in the document]
Comment (TrevorJTClarke) [reply sputnikdao-factory2/src/factory_manager.rs]: I say all this, assuming your estimations on gas all relate to default policy

[the remark this answers is not in the document]
Comment (ctindogaru) [reply sputnikdao-factory2/src/factory_manager.rs]: We can increase the Tgas to 30-40. It's still low enough to cover the new deployment costs and big enough to cover most of the policies. If a policy consumes more than 30-40, the DAO creator can create the policy slimmer in the first place and then add new roles after the DAO is created. How does that sound to you?

[the remark this answers is not in the document]
Comment (TrevorJTClarke) [reply sputnikdao-factory2/src/factory_manager.rs]: That sounds fair. Can you add a test that incorporates a large policy and update the gas to 40?

**Comment by TrevorJTClarke** [comment]
Thank you for this!!

**Pull request merged**